src/main/java/org/springframework/shell/core/JLineShellComponent.java

```diff
@@ -237,4 +238,9 @@ protected String getProductName() {
 	protected String getVersion() {
 		return version;
 	}
-}
\ No newline at end of file
+
+	public CommandLine getCommandLine() {
```

Comment:
`@VisibleForTesting` is just for documentation, so I moved the test class to the same package